BillTrack.Auth/Jwt/AuthService.cs

```diff
@@ -0,0 +1,56 @@
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using BillTrack.Core.Exceptions;
+using BillTrack.Core.Interfaces.Repositories;
+using BillTrack.Core.Interfaces.Services;
+using BillTrack.Domain.Entities;
+using FastEndpoints;
+using FastEndpoints.Security;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace BillTrack.Auth.Jwt;
+
+public class AuthService : IAuthService
+{
+    private readonly IGenericRepository<User> _userRepository;
+    private readonly IConfiguration _configuration;
+
+    public AuthService(IGenericRepository<User> userRepository, IConfiguration configuration)
+    {
+        _userRepository = userRepository;
+        _configuration = configuration;
+    }
+
+    public async Task<string> GenerateToken(string username, string password)
+    {
+        var userId = await CheckCredential(username, password);
+
+        return JwtBearer.CreateToken(
+            o =>
+            {
+                o.SigningKey = _configuration["JwtSecretKey"] ?? string.Empty;
+                o.User.Claims.Add(new Claim(JwtRegisteredClaimNames.Sub, userId.ToString()));
+                o.ExpireAt = DateTime.UtcNow.AddDays(1);
+            });
+    }
+
+    private async Task<Guid> CheckCredential(string username, string password)
```

Comment:
this method's result is user's ID. Check creds is only a part of this function - `u.Email == username && u.Password == HashPassword(password)`.
Since your code is self-describing, a function name should say about its main logic. There is a point to name it "GetUser".

Reply:
fixed